### PR TITLE
ARM32 bit fixes, for 64bit printf format specifier

### PR DESCRIPTION
**Pull request by antony-rheneus**
Signed-off-by: Antony Rheneus <arheneus@marvell.com>

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -72,7 +72,6 @@ CFLAGS_COMMON+=" -Wimport"
 CFLAGS_COMMON+=" -Winit-self"
 CFLAGS_COMMON+=" -Winline"
 CFLAGS_COMMON+=" -Winvalid-pch"
-CFLAGS_COMMON+=" -Wlong-long"
```
Comment (kcudnik) [inline configure.ac]: why removed this ?

Comment (antony-rheneus) [reply configure.ac]: ULL (unsigned long long) is valid in 32 bit arch. -Wlong-long will throw error if ULL is used.

Comment (antony-rheneus) [reply configure.ac]: for 64bit value ULL is required in 32bit arch

Comment (antony-rheneus) [reply configure.ac]: Place it is needed in SAI.
https://github.com/opencomputeproject/SAI/blob/55df00a5b4e7a89c2576e2ad3174242d948261ec/meta/saiserialize.c#L853

**Comment by lguohan** [comment]
retest this please

File: configure.ac (hunk 1 above)
Comment (kcudnik) [reply configure.ac]: its valid on 32bit arch, but not valid according to ansi standard

File: vslib/src/sai_vs_hostintf.cpp [hunk 2]
```diff
@@ -899,10 +899,9 @@ void veth2tap_fun(std::shared_ptr<hostif_info_t> info)
                 uint16_t tci = htons(aux->tp_vlan_tci);
                 uint16_t tpid = htons(IEEE_8021Q_ETHER_TYPE);
 
-                uint16_t* pvlan = (uint16_t*)(buffer + 2 * MAC_ADDRESS_SIZE);
-
-                pvlan[0] = tpid;
-                pvlan[1] = tci;
+                uint8_t* pvlan =  (uint8_t *)(buffer + 2 * MAC_ADDRESS_SIZE);
```
Comment (kcudnik) [inline vslib/src/sai_vs_hostintf.cpp]: why this is changed ?

File: meta/saiserialize.cpp [hunk 3]
```diff
@@ -2111,7 +2111,7 @@ void sai_deserialize_pointer(
 {
     SWSS_LOG_ENTER();
 
-    sai_deserialize_number(s, (uint64_t&)ptr, true);
+    sai_deserialize_number(s, (uintptr_t &)ptr, true);
```
Comment (kcudnik) [inline meta/saiserialize.cpp]: also why this is changed ?

File: vslib/src/sai_vs_hostintf.cpp (hunk 2 above)
Comment (antony-rheneus) [reply vslib/src/sai_vs_hostintf.cpp]: cast-align warnings, as the char * lesser size pointer is not allowed to be casted to uint_16 * bigger pointer. 
Error:-
cast increases required alignment of target type [-Wcast-align]

File: meta/saiserialize.cpp (hunk 3 above)
Comment (antony-rheneus) [reply meta/saiserialize.cpp]: same cast-align warning, cannot convert a pointer to a 64 bit as size is increased.

File: vslib/src/sai_vs_hostintf.cpp (hunk 2 above)
Comment (kcudnik) [reply vslib/src/sai_vs_hostintf.cpp]: how you compile this since so far i didnt get such errors

Comment (antony-rheneus) [reply vslib/src/sai_vs_hostintf.cpp]:
```
Linux et6448m-M0 4.14.76-devel-18.12.2-00957-gef92281-dirty #3 SMP Mon Jul 8 01:26:28 PDT 2019 armv7l GNU/Linux
---
cat /proc/cpuinfo
processor       : 0
model name      : ARMv7 Processor rev 1 (v7l)
BogoMIPS        : 50.00
Features        : half thumb fastmult vfp edsp thumbee neon vfpv3 tls vfpd32
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x4
CPU part        : 0xc09
CPU revision    : 1

processor       : 1
model name      : ARMv7 Processor rev 1 (v7l)
BogoMIPS        : 50.00
Features        : half thumb fastmult vfp edsp thumbee neon vfpv3 tls vfpd32
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x4
CPU part        : 0xc09
CPU revision    : 1

Hardware        : Marvell Armada 380/385 (Device Tree)
Revision        : 0000
Serial          : 0000000000000000
---
$ gcc -dumpmachine
arm-linux-gnueabihf
----
$ gcc -v
Using built-in specs.
COLLECT_GCC=/usr/bin/gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/arm-linux-gnueabihf/6/lto-wrapper
Target: arm-linux-gnueabihf
Configured with: ../src/configure -v --with-pkgversion='Debian 6.3.0-18+deb9u1' --with-bugurl=file:///usr/share/doc/gcc-6/README.Bugs --enable-languages=c,ada,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-6 --program-prefix=arm-linux-gnueabihf- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-libitm --disable-libquadmath --enable-plugin --enable-default-pie --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-6-armhf/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-6-armhf --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-6-armhf --with-arch-directory=arm --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --with-target-system-zlib --enable-objc-gc=auto --enable-multiarch --disable-sjlj-exceptions --with-arch=armv7-a --with-fpu=vfpv3-d16 --with-float=hard --with-mode=thumb --enable-checking=release --build=arm-linux-gnueabihf --host=arm-linux-gnueabihf --target=arm-linux-gnueabihf
Thread model: posix
gcc version 6.3.0 20170516 (Debian 6.3.0-18+deb9u1)

```

Comment (kcudnik) [reply vslib/src/sai_vs_hostintf.cpp]: you could try do this on more recent version ?

**Comment by lguohan** [comment]
can you resolve the conflict?

File: configure.ac (hunk 1 above)
Comment (antony-rheneus) [reply configure.ac]: Need this to support 32bit arch compilation

File: vslib/src/sai_vs_hostintf.cpp (hunk 2 above)
Comment (antony-rheneus) [reply vslib/src/sai_vs_hostintf.cpp]: Debian stretch provides, gcc 6.3 version

**Pull request merged**